src/par_iter/mod.rs

```diff
@@ -609,18 +609,19 @@ pub trait ParallelIterator: Sized {
     fn drive_unindexed<C>(self, consumer: C) -> C::Result
         where C: UnindexedConsumer<Self::Item>;
 
+    /// Internal method used to fake specialization for indexed collect.
+    #[doc(hidden)]
+    fn opt_len(&mut self) -> Option<usize> { None }
+
     /// Create a fresh collection containing all the element produced
-    /// by this parallel iterator. Note that some kinds of collections
-    /// have stricter requirements in terms of the kinds of iterators
-    /// that you can collect from (e.g., a `Vec` currently requires an
-    /// iterator that has precise knowledge of how many elements it
-    /// contains).
-    ///
-    /// You may also prefer to use `collect_into()`, which allows you
-    /// to reuse the vector's backing store rather than allocating a
-    /// fresh vector.
+    /// by this parallel iterator.
+    ///
+    /// You may prefer to use `collect_into()`, which allocates more
+    /// efficiently with precise knowledge of how many elements the
+    /// iterator contains, and even allows you to reuse an existing
+    /// vector's backing store rather than allocating a fresh vector.
```

Comment:
Do you think we can make `collect_into()` work for any collection `C` where `C: FromParallelIterator`? I guess we'd have to rework the `FromParallelIterator` trait somewhat to do so. Let's leave it as is for now.